File: AICodingWeb/aicoding_web/app/models/stock_item.py
from datetime import datetime
from app import db
from app.models.base import BaseModel

class StockItem(db.Model, BaseModel):
    """StockItem model"""
    
    __tablename__ = 'StockItems'
    __table_args__ = {'schema': 'Warehouse'}
    
    StockItemID = db.Column(db.Integer, primary_key=True)
    StockItemName = db.Column(db.String(100), nullable=False)
    SupplierID = db.Column(db.Integer, db.ForeignKey('Purchasing.Suppliers.SupplierID'), nullable=False)
    ColorID = db.Column(db.Integer, db.ForeignKey('Warehouse.Colors.ColorID'))
    UnitPackageID = db.Column(db.Integer, db.ForeignKey('Warehouse.PackageTypes.PackageTypeID'), nullable=False)
    OuterPackageID = db.Column(db.Integer, db.ForeignKey('Warehouse.PackageTypes.PackageTypeID'), nullable=False)
    Brand = db.Column(db.String(50))
    Size = db.Column(db.String(20))
    LeadTimeDays = db.Column(db.Integer, nullable=False)
    QuantityPerOuter = db.Column(db.Integer, nullable=False)
    IsChillerStock = db.Column(db.Boolean, nullable=False)
    Barcode = db.Column(db.String(50))
    TaxRate = db.Column(db.Numeric(18, 3), nullable=False)
    UnitPrice = db.Column(db.Numeric(18, 2), nullable=False)
    RecommendedRetailPrice = db.Column(db.Numeric(18, 2))
    TypicalWeightPerUnit = db.Column(db.Numeric(18, 3), nullable=False)
    MarketingComments = db.Column(db.Text)
    InternalComments = db.Column(db.Text)
    Photo = db.Column(db.LargeBinary)
    CustomFields = db.Column(db.JSON)
    Tags = db.Column(db.JSON)
    SearchDetails = db.Column(db.Text, nullable=False)
    LastEditedBy = db.Column(db.Integer, db.ForeignKey('Application.People.PersonID'), nullable=False)
    ValidFrom = db.Column(db.DateTime, nullable=False)
    ValidTo = db.Column(db.DateTime, nullable=False)
    
    def __repr__(self):
        return f'<StockItem {self.StockItemID}>' 

File: AICodingWeb/Models/StockItem.cs
using System;
using System.Collections.Generic;

namespace AICodingWeb.Models
{
    public class StockItem
    {
        public int StockItemID { get; set; }
        public string StockItemName { get; set; }
        public int SupplierID { get; set; }
        public int? ColorID { get; set; }
        public int UnitPackageID { get; set; }
        public int OuterPackageID { get; set; }
        public string Brand { get; set; }
        public string Size { get; set; }
        public int LeadTimeDays { get; set; }
        public int QuantityPerOuter { get; set; }
        public bool IsChillerStock { get; set; }
        public string Barcode { get; set; }
        public decimal TaxRate { get; set; }
        public decimal UnitPrice { get; set; }
        public decimal? RecommendedRetailPrice { get; set; }
        public decimal TypicalWeightPerUnit { get; set; }
        public string MarketingComments { get; set; }
        public string InternalComments { get; set; }
        public byte[] Photo { get; set; }
        public string CustomFields { get; set; }
        public string Tags { get; set; }
        public string SearchDetails { get; set; }
        public int LastEditedBy { get; set; }
        public DateTime LastEditedWhen { get; set; }

        public virtual Supplier Supplier { get; set; }
        public virtual Color Color { get; set; }
        public virtual PackageType UnitPackage { get; set; }
        public virtual PackageType OuterPackage { get; set; }
        public virtual ICollection<InvoiceLine> InvoiceLines { get; set; }
        public virtual ICollection<OrderLine> OrderLines { get; set; }
        public virtual ICollection<PurchaseOrderLine> PurchaseOrderLines { get; set; }
        public virtual ICollection<SpecialDeal> SpecialDeals { get; set; }
        public virtual ICollection<StockItemStockGroup> StockItemStockGroups { get; set; }
        public virtual ICollection<StockItemTransaction> StockItemTransactions { get; set; }
    }
} 